### PR TITLE
Use a Vue directive for event listening

### PR DESCRIPTION
**Pull request by rictic**
@justinfagnani would appreciate your review too

I'm a bit torn about whether to count this towards Vue's score. On the one hand, it's custom imperative code, and of course any framework can work with custom imperative code. On the other hand, it's a tiny amount of code, and with it, it's just as easy and ergonomic as the built-in `v-on` syntax.

### CODE REVIEW

**Comment by rictic** [comment]
See discussion at https://github.com/vuejs/core/issues/5401#issuecomment-1041214293

**Review by AndrewJakubowicz — APPROVED** [review]
Minor nits. Left an opinion but nothing blocking,

I agree that a custom directive is not too bad. Since once you set it up it becomes declarative.

File: libraries/vue/meta/issues.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [
   {
     "link": "https://github.com/vuejs/core/issues/5401",
-    "title": "Vue 3 does not support listening to custom events with capital letters",
+    "title": "Vue 3's v-on syntax does not support listening to custom events with capital letters",
```
[the remark this answers is not in the document]
Comment (rictic) [reply libraries/vue/meta/issues.json:4]: I think it's probably good to keep the issue visible, because if you just do the obvious thing it won't work.

**Comment by justinfagnani** [comment]
I don't think this should count as 100% on this particular test. Most frameworks can interoperate with APIs ones you drop to imperative code. React can use refs to set properties and listen for events. It help if you can wrap the imperative code in a reusable abstraction, but it's still a potential cost to using an element. I'm actually not aware of a common framework where you _couldn't_ use some feature with an imperative workaround.

That said, I'm not sure that all these tests should be weighted equally. Some of these are edge or rarer cases here for completeness. I don't know how we'd estimate this well, but I'm sure that lower and kebab-case event names are the vast majority of event names (though MWC attempted to use mixed-case event names for a bit?). It doesn't seem right that a mixed-case event test should carry near the same weight as kebab-case or being able to set properties.

**Comment by rictic** [comment]
> React can use refs to set properties and listen for events.

Yeah, but if there was an 8-line bit of code you could drop into any React app to get a great developer experience for working with props and events in React, that'd be very different than the status quo, yeah?

Like, it's not great that you can't use `v-on`, but `v-event` is very chill, and you can add it at the application level too.

I suppose that's a pretty good argument for adding it as a built in directive rather than making it custom though, so that it's standard and easy to google at least.

**Comment by edimitchel** [comment]
This is now fixed since the version 3.2.38 of Vue.

**Comment by rictic** [comment]
Super happy with the fix on the vue side!

**Pull request closed without merging**